types/tx.go

```diff
@@ -69,7 +70,7 @@ func (txs Txs) Proof(i int) TxProof {
 	l := len(txs)
 	bzs := make([][]byte, l)
 	for i := 0; i < l; i++ {
-		bzs[i] = txs[i]
+		bzs[i] = txs[i].Hash()
```

Comment:
Think we need to update this in the spec. It seems txs are treated differently from everything else by being hashed before passed to the MerkleTree. Should the other data structures be treated similarly, so they're hashed before being passed in (and eg. concatenated with 0) ?